### PR TITLE
Added option to install under systemd rather than upstart

### PR DESCRIPTION
**Pull request by ncasaril**
Sorry for the "hackyness" - I wanted a quick fix to allow installation under systemd. Should perhaps be under a kinetic branch as well. 

This links to issue #37 

Tested under ubuntu 16.04 on x86_64 and armv7 with Kinetic. Feedback appreciated :)

### CODE REVIEW

File: src/robot_upstart/install_script.py [hunk 1]
```diff
@@ -91,6 +94,10 @@ def main():
     if args.augment:
         j.generate_system_files = False
 
-    j.install()
+    provider=providers.Upstart
+    if args.systemd:
+        provider=providers.Systemd
+
+    j.install(Provider=provider)
```
Comment (mikepurvis) [inline src/robot_upstart/install_script.py:112]: Ideally we would detect the init system here and just do the right thing, but having looked into it previously, I acknowledge that detecting it reliably can be tricky. Other possibility for an interface would be `--provider systemd`, but I'm not super attached to that; the flag is fine.

**Comment by ncasaril** [comment]
I'm not entirely satisfied that my solution actually enables start at boot
time. Leave it with me and I'll run some more test and verify and/or fix
the startup behaviour.

**Comment by ncasaril** [comment]
The final change where I added a post_install function to the providers should probably either be creating the required symlink:

```
/etc/systemd/system/multi-user.target.wants/(job_name).service -> /lib/systemd/system/(job_name).service
```

or we need to have the functionality to also run commands as sudo through a similar method as the mutate_files script.

**Comment by mikepurvis** [comment]
Note that #43 is making the `mutate_files` script able to create symlinks.

**Comment by mikepurvis** [comment]
This looks great, thanks for taking a stab at the detection thing too. Let me know if you want to look in the symlink thing. Either way, if this is ready for merge, can I trouble you to rebase?

**Comment by ncasaril** [comment]
I'll have a look at the symlink possibility as this would remove the need
for the post function altogether. And then I'll rebase as well. :)

**Comment by ncasaril** [comment]
I actually merged in the latest changes from your jade-devel branch instead of rebasing - was that a mistake? 
Sorry but I'm very familiar with the rebase part of git. Would you mind pointing me in the right direction?

**Comment by mikepurvis** [comment]
No worries, that'll do!

**Comment by mikepurvis** [comment]
For future reference: https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Pull request merged**